backstory/src/main/java/com/google/sps/perspective/PerspectiveManager.java

```diff
@@ -48,10 +55,21 @@
   /**
    * Constructs an object which implements the StoryAnalysisManager
    * class using the Google Perspective API.
+   *
+   * @throws APINotAvailableException when it can't create an instance of the PerspectiveAPI
+   *    (this most likely occurs if the "PerspectiveAPIKey.java" file is not present)
    */
-  public PerspectiveManager() throws NoAppropriateStoryException {
-    // TODO: instantiate the PerspectiveAPI for this instance
-    perspectiveAPI = null;
+  public PerspectiveManager() throws APINotAvailableException {
+    PerspectiveAPIFactory factory;
+
+    try {
+      factory = new PerspectiveAPIFactoryImpl();
+    } catch (ClassNotFoundException | NoSuchMethodException | IllegalAccessException
```

Comment:
Blanketing all these different exceptions as a single APINotAvailable exception may frustrating for a user. Perhaps return the specific exception name in the exception string?

Reply:
I'm including the exception error message in the APINotAvailable error message. @ykjchen Do you think this is sufficient?